### PR TITLE
chore(cli): optimize bundle size

### PR DESCRIPTION
**Pull request by 0xmad**
# Description

Use separate tsconfig for build and development

## Additional Notes

N/A

## Related issue(s)

Related to #781 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://github.com/privacy-scaling-explorations/maci/blob/dev/CONTRIBUTING.md) and [code of conduct](https://github.com/privacy-scaling-explorations/maci/blob/dev/CODE_OF_CONDUCT.md).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

**Review by ctrlc03 — APPROVED** [review]
Thank you! Is this something we can (and should) do to other packages if we move out tests from the ts folder?

**Comment by 0xmad** [comment]
@ctrlc03 I think I can exclude `*.test.ts` files and we don't have any of tests in any build folder.

File: circuits/tsconfig.build.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build"
+  },
+  "include": ["./ts"],
+  "exclude": ["**/__tests__/**", "**/__tests__/**/*.test.ts", "**/__tests__/**/*.test.ts"]
```
Comment (ctrlc03) [inline circuits/tsconfig.build.json:7]: Looks like `"**/__tests__/**/*.test.ts"`  is repeated twice

File: cli/tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build"
+  },
+  "include": ["./ts"],
+  "exclude": ["**/__tests__/**", "**/__tests__/**/*.test.ts", "**/__tests__/**/*.test.ts"],
```
Comment (ctrlc03) [inline cli/tsconfig.build.json:7]: shouldn't be needed here as all tests are outside the ts folder

File: contracts/tsconfig.build.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build"
+  },
+  "include": ["./ts", "./typechain-types"],
+  "exclude": ["**/__tests__/**", "**/__tests__/**/*.test.ts", "**/__tests__/**/*.test.ts"],
```
Comment (ctrlc03) [inline contracts/tsconfig.build.json:7]: same here the tests are in their own folder

File: domainobjs/tsconfig.build.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./build"
+  },
+  "include": ["./ts"],
+  "exclude": ["**/__tests__/**", "**/__tests__/**/*.test.ts", "**/__tests__/**/*.test.ts"]
```
Comment (ctrlc03) [inline domainobjs/tsconfig.build.json:7]: "**/__tests__/**/*.test.ts" is repeated

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Just a couple of comments! thank you

**Review by ctrlc03 — APPROVED** [review]
Thank you!!

**Pull request merged**